packages/flutter_tools/lib/src/build_info.dart

```diff
@@ -113,3 +115,23 @@ HostPlatform getCurrentHostPlatform() {
 
   return HostPlatform.linux_x64;
 }
+
+/// Returns the top-level build output directory.
+String getBuildDirectory() {
+  return config.getValue('build-dir') ?? 'build';
```

Comment:
Yeah; asked that in the first-round comments but that's two of us; I'll add an assertion on `!path.isAbsolute()`